pallets/maintenance-mode/src/lib.rs

```diff
@@ -0,0 +1,157 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+/// Edit this file to define custom logic or remove it if it is not needed.
+/// Learn more about FRAME and the core library of Substrate FRAME pallets:
+/// <https://docs.substrate.io/reference/frame-pallets/>
+pub use pallet::*;
+
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarking;
+pub mod weights;
+pub use weights::*;
+
+pub const ACTIVATED: bool = true;
+pub const DEACTIVATEDD: bool = false;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use cumulus_primitives_core::{
+		relay_chain::BlockNumber as RelayBlockNumber, DmpMessageHandler,
+	};
+	use frame_support::{
+		pallet_prelude::{ValueQuery, *},
+		traits::Contains,
+	};
+	use frame_system::pallet_prelude::*;
+	pub use log;
+	use sp_std::vec::Vec;
+	use xcm_primitives::PauseXcmExecution;
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T>(_);
+
+	#[derive(Default)]
+	#[pallet::genesis_config]
+	/// Genesis config for maintenance mode pallet
+	pub struct GenesisConfig {}
+
+	#[pallet::genesis_build]
+	impl<T: Config> GenesisBuild<T> for GenesisConfig {
+		fn build(&self) {
+			MaintenanceModeStatus::<T>::put(ACTIVATED);
+		}
+	}
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+		type MaintenanceModeOrigin: EnsureOrigin<Self::RuntimeOrigin>;
+		type FilteredCalls: Contains<Self::RuntimeCall>;
+		type MaintenanceDmpHandler: DmpMessageHandler;
+		type XcmExecutorManager: PauseXcmExecution;
+		type WeightInfo: WeightInfo;
+	}
+
+	#[pallet::storage]
+	pub type MaintenanceModeStatus<T: Config> = StorageValue<_, bool, ValueQuery>;
```

Comment:
is a valid comment, however my personal preference is to keep using the `::get()` instead of the getter. I think that 
`MaintenanceModeStatus::<T>::get()` gives more context than `Self::is_active()`. Also i have seen a lot of getters not used in many pallets.

Reply:
I feel like both approaches are good for different reasons. `Self::is_active()` is cleaner  in terms of code ergonomics and you can automatically assume that it will return a `boolean`, but in order to understand it you either have to read the hole line to get extra context or go to the definition of the function.

Whereas `MaintenanceModeStatus::<T>::get()` is pretty self explainatory, but not as clean.

I think is certainly a matter of taste here.